File: debian.py
#!/usr/bin/env python

import os

import requests
from bs4 import BeautifulSoup


def find_debian_version(release):
    '''Fetch a complete Debian release.'''

    r = requests.get(f'https://cdimage.debian.org/cdimage/{release}/amd64/iso-cd')
    r.raise_for_status()
    s = BeautifulSoup(r.text, 'html.parser')

    for link in s.find_all('a'):
        if 'iso' in link.text and 'netinst' in link.text and 'mac' not in link.text and 'edu' not in link.text:
            iso = link.text

    print(f'{iso}')


def main():
    '''Main function.'''

    find_debian_version('release/current')
    find_debian_version('archive/latest-oldstable')


if __name__ == '__main__':
    main()

File: debian.go
package main

import (
	"fmt"
	"log"
	"net/http"
	"strings"

	"github.com/PuerkitoBio/goquery"
)

func dumpOne(url string) {
	res, err := http.Get(url)
	if err != nil {
		log.Fatal(err)
	}
	defer res.Body.Close()
	if res.StatusCode != 200 {
		log.Fatalf("Status code error: %d %s", res.StatusCode, res.Status)
	}

	doc, err := goquery.NewDocumentFromReader(res.Body)
	if err != nil {
		log.Fatal("Error loading HTTP response body.", err)
	}

	doc.Find("a").Each(func(index int, element *goquery.Selection) {
		href, exists := element.Attr("href")
		if exists {
			if strings.Contains(href, "netinst.iso") || strings.Contains(href, "SHA") {
				fmt.Println(fmt.Sprintf("%s%s", url, href))
			}
		}
	})
}

func main() {
  const (
    testing      = "http://cdimage.debian.org/cdimage/weekly-builds/arm64/iso-cd/"
    stable       = "http://cdimage.debian.org/cdimage/release/current/amd64/iso-cd/"
    oldstable    = "http://cdimage.debian.org/cdimage/archive/latest-oldstable/amd64/iso-cd/"
    oldoldstable = "http://cdimage.debian.org/cdimage/archive/latest-oldoldstable/amd64/iso-cd/"
  )

  dumpOne(testing)
  dumpOne(stable)
  dumpOne(oldstable)
  dumpOne(oldoldstable)
}
